src/SdfGenerator.cc

```diff
@@ -421,16 +441,254 @@ namespace sdf_generator
+  {
+    // Update sdf based on current components. Here are the list of components
+    // to be updated:
+    // - Name
+    // - Pose
+    // - Inertial
+    // - WindMode
+    // This list is to be updated as other components become updateable during
+    // simulation
+    auto *nameComp = _ecm.Component<components::Name>(_entity);
+    _elem->GetAttribute("name")->Set(nameComp->Data());
+
+    auto *poseComp = _ecm.Component<components::Pose>(_entity);
+
+    auto poseElem = _elem->GetElement("pose");
+
+    // Remove all attributes of poseElem
+    for (const auto *attrName : {"relative_to", "degrees", "rotation_format"})
     {
-      // Update relative URIs to use absolute paths. Relative URIs work fine in
-      // included models, but they have to be converted to absolute URIs when
-      // the included model is expanded.
-      relativeToAbsoluteUri(_elem, common::parentPath(pathComp->Data()));
+      sdf::ParamPtr attr = poseElem->GetAttribute(attrName);
+      if (nullptr != attr)
+      {
+        attr->Reset();
+      }
     }
+    poseElem->Set(poseComp->Data());
+
+    // inertial
+    auto inertialComp = _ecm.Component<components::Inertial>(_entity);
+    if (inertialComp)
+    {
+      math::Inertiald inertial = inertialComp->Data();
+      sdf::ElementPtr inertialElem = _elem->GetElement("inertial");
+      inertialElem->GetElement("pose")->Set<math::Pose3d>(inertial.Pose());
+      const math::MassMatrix3d &massMatrix = inertial.MassMatrix();
+      inertialElem->GetElement("mass")->Set<double>(massMatrix.Mass());
+      sdf::ElementPtr inertiaElem = inertialElem->GetElement("inertia");
+      inertiaElem->GetElement("ixx")->Set<double>(massMatrix.Ixx());
+      inertiaElem->GetElement("ixy")->Set<double>(massMatrix.Ixy());
+      inertiaElem->GetElement("ixz")->Set<double>(massMatrix.Ixz());
+      inertiaElem->GetElement("iyy")->Set<double>(massMatrix.Iyy());
+      inertiaElem->GetElement("iyz")->Set<double>(massMatrix.Iyz());
+      inertiaElem->GetElement("izz")->Set<double>(massMatrix.Izz());
+    }
+
+    // wind mode
+    auto windModeComp = _ecm.Component<components::WindMode>(_entity);
+    if (windModeComp)
+    {
+      bool windMode = windModeComp->Data();
+      _elem->GetElement("enable_wind")->Set<bool>(windMode);
+    }
+
+    // update sensors
+    if (_elem->HasElement("sensor"))
+    {
+      sdf::ElementPtr sensorElem = _elem->GetElement("sensor");
+      while (sensorElem)
+      {
+        std::string sensorName = sensorElem->Get<std::string>("name");
+        auto sensorEnt = _ecm.EntityByComponents(
+            components::ParentEntity(_entity), components::Name(sensorName));
+        if (sensorEnt != kNullEntity)
+          updateSensorElement(sensorElem, _ecm, sensorEnt);
+        sensorElem = sensorElem->GetNextElement("sensor");
+      }
+    }
+
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  bool updateSensorElement(sdf::ElementPtr _elem,
```

Comment:
Would it be possible to use the `sdf::Sensor` class along with its `ToElement` function?

Reply:
I'm working on `ToElement` functions for the sdf Model, Link, Visual, and Collision classes. This should help reduce the code in ign-gazebo.

Reply:
> Would it be possible to use the sdf::Sensor class along with its ToElement function?

yeah it's should be using the `sdf::Sensor::ToElement()` functions below. The various `if` checks below are for getting the correct sensor component from the entity. I also override the name and pose based on latest values from the ECM - not sure if it's really needed.

Reply:
Thanks. I'm shelving my `ToElement` additions. It's too many changes given the time line.